python/ray/experimental/workflow/storage.py

```diff
@@ -0,0 +1,94 @@
+import json
+import pathlib
+from typing import Any, Union
+import uuid
+
+import ray
+import ray.cloudpickle
+import ray._private.utils
+
+# TODO(suquark): in the future we may use general URLs to support
+# different storages, e.g. "s3://xxxx/xxx". Currently we just use
+# 'pathlib.Path' for convenience.
+
+
+class Storage:
```

Comment:
Can you clean up this interface so it doesn't expose the implementation detail of `Path`?